From 29ff9a02c767cf7c2c46c808aca906e7ab4a90e5 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Mon, 22 Jan 2024 06:18:11 +0100
Subject: [PATCH] media-resource-calculator: Add generic config files

So we can build it on every target even though the config might not 100% correct, at least it can build.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

---
 calculator/CMakeLists.txt          | 13 +++-
 calculator/resource_calculator.cpp |  5 +-
 data/generic_adec.json             |  5 ++
 data/generic_disp.json             |  3 +
 data/generic_vdec.json             | 97 ++++++++++++++++++++++++++++++
 data/generic_venc.json             | 78 ++++++++++++++++++++++++
 6 files changed, 199 insertions(+), 2 deletions(-)
 create mode 100644 data/generic_adec.json
 create mode 100644 data/generic_disp.json
 create mode 100644 data/generic_vdec.json
 create mode 100644 data/generic_venc.json

diff --git a/calculator/CMakeLists.txt b/calculator/CMakeLists.txt
index 8a2dcca..f5aafd7 100644
--- a/calculator/CMakeLists.txt
+++ b/calculator/CMakeLists.txt
@@ -82,7 +82,18 @@ elseif(${WEBOS_TARGET_MACHINE} MATCHES "qemux86" OR ${WEBOS_TARGET_MACHINE} MATC
   set(DISP_RESOURCE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/disp_resources_qemux86.h)
 
 else()
-  message("Check the resource calculator json files")
+  message("Build resource calculator for generic platform")
+
+  add_definitions(-DPLATFORM_GENERIC)
+
+  set(ADEC_RESOURCE_JSON ../data/generic_adec.json)
+  set(VDEC_RESOURCE_JSON ../data/generic_vdec.json)
+  set(VENC_RESOURCE_JSON ../data/generic_venc.json)
+  set(DISP_RESOURCE_JSON ../data/generic_disp.json)
+  set(ADEC_RESOURCE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/adec_resources_generic.h)
+  set(VDEC_RESOURCE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/vdec_resources_generic.h)
+  set(VENC_RESOURCE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/venc_resources_generic.h)
+  set(DISP_RESOURCE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/disp_resources_generic.h)
 endif()
 
 # Generate resource headers from resource json files
diff --git a/calculator/resource_calculator.cpp b/calculator/resource_calculator.cpp
index 6e6acb5..af1fd0a 100644
--- a/calculator/resource_calculator.cpp
+++ b/calculator/resource_calculator.cpp
@@ -43,7 +43,10 @@
 #include "disp_resources_qemux86.h"
 
 #else
-#  error "platform is not specified"
+#include "adec_resources_generic.h"
+#include "vdec_resources_generic.h"
+#include "venc_resources_generic.h"
+#include "disp_resources_generic.h"
 #endif
 
 namespace mrc {
diff --git a/data/generic_adec.json b/data/generic_adec.json
new file mode 100644
index 0000000..3b33828
--- /dev/null
+++ b/data/generic_adec.json
@@ -0,0 +1,5 @@
+{
+  "default": {
+    "ADEC": 1
+  }
+}
diff --git a/data/generic_disp.json b/data/generic_disp.json
new file mode 100644
index 0000000..c0305ba
--- /dev/null
+++ b/data/generic_disp.json
@@ -0,0 +1,3 @@
+{
+  "DISP": {"DISP": 1}
+}
diff --git a/data/generic_vdec.json b/data/generic_vdec.json
new file mode 100644
index 0000000..4e7d2c9
--- /dev/null
+++ b/data/generic_vdec.json
@@ -0,0 +1,97 @@
+{
+  "H265": {
+    "1280*720 0p": { "VDEC": 8 },
+    "1280*720 24p": { "VDEC": 8 },
+    "1280*720 30p": { "VDEC": 8 },
+    "1280*720 60p": { "VDEC": 8 },
+    "1280*720 120p": { "VDEC": 8 },
+    "1920*1080 0p": { "VDEC": 8 },
+    "1920*1080 24p": { "VDEC": 8 },
+    "1920*1080 30p": { "VDEC": 8 },
+    "1920*1080 60p": { "VDEC": 8 },
+    "3840*2160 0p": { "VDEC": 8 },
+    "3840*2160 24p": { "VDEC": 8 },
+    "3840*2160 30p": { "VDEC": 8 },
+    "4096*2160 0p": { "VDEC": 8 },
+    "4096*2160 24p": { "VDEC": 8 },
+    "4096*2160 30p": { "VDEC": 8 },
+    "4096*2304 0p": { "VDEC": 8 },
+    "4096*2304 15p": { "VDEC": 8 }
+  },
+  "H264": {
+    "1280*720 24p": { "VDEC": 8 },
+    "1280*720 30p": { "VDEC": 8 },
+    "1280*720 60p": { "VDEC": 8 },
+    "1280*720 120p": { "VDEC": 8 },
+    "1280*720 0p": { "VDEC": 8},
+    "1920*1080 24p": { "VDEC": 8 },
+    "1920*1080 30p": { "VDEC": 8 },
+    "1920*1080 60p": { "VDEC": 8 },
+    "1920*1080 0p": { "VDEC": 8 },
+    "3840*2160 0p": { "VDEC": 8 },
+    "3840*2160 24p": { "VDEC": 8 },
+    "3840*2160 30p": { "VDEC": 8 },
+    "4096*2160 0p": { "VDEC": 8 },
+    "4096*2160 24p": { "VDEC": 8 },
+    "4096*2160 30p": { "VDEC": 8 },
+    "4096*2304 0p": { "VDEC": 8 },
+    "4096*2304 15p": { "VDEC": 8 }
+  },
+  "VP8": {
+    "1280*720 0p": { "VDEC": 8 },
+    "1280*720 24p": { "VDEC": 8 },
+    "1280*720 30p": { "VDEC": 8 },
+    "1280*720 60p": { "VDEC": 8 },
+    "1280*720 120p": { "VDEC": 8 },
+    "1920*1080 0p": { "VDEC": 8 },
+    "1920*1080 24p": { "VDEC": 8 },
+    "1920*1080 30p": { "VDEC": 8 },
+    "1920*1080 60p": { "VDEC": 8 },
+    "3840*2160 0p": { "VDEC": 8 },
+    "3840*2160 24p": { "VDEC": 8 },
+    "3840*2160 30p": { "VDEC": 8 },
+    "4096*2160 0p": { "VDEC": 8 },
+    "4096*2160 24p": { "VDEC": 8 },
+    "4096*2160 30p": { "VDEC": 8 },
+    "4096*2304 0p": { "VDEC": 8 },
+    "4096*2304 15p": { "VDEC": 8 }
+  },
+  "VP9": {
+    "1280*720 0p": { "VDEC": 8 },
+    "1280*720 24p": { "VDEC": 8 },
+    "1280*720 30p": { "VDEC": 8 },
+    "1280*720 60p": { "VDEC": 8 },
+    "1280*720 120p": { "VDEC": 8 },
+    "1920*1080 0p": { "VDEC": 8 },
+    "1920*1080 24p": { "VDEC": 8 },
+    "1920*1080 30p": { "VDEC": 8 },
+    "1920*1080 60p": { "VDEC": 8 },
+    "3840*2160 0p": { "VDEC": 8 },
+    "3840*2160 24p": { "VDEC": 8 },
+    "3840*2160 30p": { "VDEC": 8 },
+    "4096*2160 0p": { "VDEC": 8 },
+    "4096*2160 24p": { "VDEC": 8 },
+    "4096*2160 30p": { "VDEC": 8 },
+    "4096*2304 0p": { "VDEC": 8 },
+    "4096*2304 15p": { "VDEC": 8 }
+  },
+  "default": {
+    "1280*720 24p": { "VDEC": 8 },
+    "1280*720 30p": { "VDEC": 8 },
+    "1280*720 60p": { "VDEC": 8 },
+    "1280*720 120p": { "VDEC": 8 },
+    "1280*720 0p": { "VDEC": 8},
+    "1920*1080 24p": { "VDEC": 8 },
+    "1920*1080 30p": { "VDEC": 8 },
+    "1920*1080 60p": { "VDEC": 8 },
+    "1920*1080 0p": { "VDEC": 8 },
+    "3840*2160 0p": { "VDEC": 8 },
+    "3840*2160 24p": { "VDEC": 8 },
+    "3840*2160 30p": { "VDEC": 8 },
+    "4096*2160 0p": { "VDEC": 8 },
+    "4096*2160 24p": { "VDEC": 8 },
+    "4096*2160 30p": { "VDEC": 8 },
+    "4096*2304 0p": { "VDEC": 8 },
+    "4096*2304 15p": { "VDEC": 8 }
+  }
+}
diff --git a/data/generic_venc.json b/data/generic_venc.json
new file mode 100644
index 0000000..6b16700
--- /dev/null
+++ b/data/generic_venc.json
@@ -0,0 +1,78 @@
+{
+  "H264": {
+    "1280*720 0p": { "VENC": 2 },
+    "1280*720 24p": { "VENC": 1 },
+    "1280*720 30p": { "VENC": 1 },
+    "1280*720 60p": { "VENC": 1 },
+    "1280*720 120p": { "VENC": 2 },
+    "1920*1080 0p": { "VENC": 2 },
+    "1920*1080 24p": { "VENC": 1 },
+    "1920*1080 30p": { "VENC": 1 },
+    "1920*1080 60p": { "VENC": 2 },
+    "3840*2160 0p": { "VENC": 4 },
+    "3840*2160 24p": { "VENC": 4 },
+    "3840*2160 30p": { "VENC": 4 },
+    "4096*2160 0p": { "VENC": 5 },
+    "4096*2160 24p": { "VENC": 5 },
+    "4096*2160 30p": { "VENC": 5 },
+    "4096*2304 0p": { "VENC": 8 },
+    "4096*2304 15p": { "VENC": 8 }
+  },
+  "H265": {
+    "1280*720 0p": { "VENC": 2 },
+    "1280*720 24p": { "VENC": 1 },
+    "1280*720 30p": { "VENC": 1 },
+    "1280*720 60p": { "VENC": 1 },
+    "1280*720 120p": { "VENC": 2 },
+    "1920*1080 0p": { "VENC": 2 },
+    "1920*1080 24p": { "VENC": 1 },
+    "1920*1080 30p": { "VENC": 1 },
+    "1920*1080 60p": { "VENC": 2 },
+    "3840*2160 0p": { "VENC": 4 },
+    "3840*2160 24p": { "VENC": 4 },
+    "3840*2160 30p": { "VENC": 4 },
+    "4096*2160 0p": { "VENC": 5 },
+    "4096*2160 24p": { "VENC": 5 },
+    "4096*2160 30p": { "VENC": 5 },
+    "4096*2304 0p": { "VENC": 8 },
+    "4096*2304 15p": { "VENC": 8 }
+  },
+  "VP8": {
+    "1280*720 0p": { "VENC": 2 },
+    "1280*720 24p": { "VENC": 1 },
+    "1280*720 30p": { "VENC": 1 },
+    "1280*720 60p": { "VENC": 1 },
+    "1280*720 120p": { "VENC": 2 },
+    "1920*1080 0p": { "VENC": 2 },
+    "1920*1080 24p": { "VENC": 1 },
+    "1920*1080 30p": { "VENC": 1 },
+    "1920*1080 60p": { "VENC": 2 },
+    "3840*2160 0p": { "VENC": 4 },
+    "3840*2160 24p": { "VENC": 4 },
+    "3840*2160 30p": { "VENC": 4 },
+    "4096*2160 0p": { "VENC": 5 },
+    "4096*2160 24p": { "VENC": 5 },
+    "4096*2160 30p": { "VENC": 5 },
+    "4096*2304 0p": { "VENC": 8 },
+    "4096*2304 15p": { "VENC": 8 }
+  },
+  "default": {
+    "1280*720 0p": { "VENC": 2 },
+    "1280*720 24p": { "VENC": 1 },
+    "1280*720 30p": { "VENC": 1 },
+    "1280*720 60p": { "VENC": 1 },
+    "1280*720 120p": { "VENC": 2 },
+    "1920*1080 0p": { "VENC": 2 },
+    "1920*1080 24p": { "VENC": 1 },
+    "1920*1080 30p": { "VENC": 1 },
+    "1920*1080 60p": { "VENC": 2 },
+    "3840*2160 0p": { "VENC": 4 },
+    "3840*2160 24p": { "VENC": 4 },
+    "3840*2160 30p": { "VENC": 4 },
+    "4096*2160 0p": { "VENC": 5 },
+    "4096*2160 24p": { "VENC": 5 },
+    "4096*2160 30p": { "VENC": 5 },
+    "4096*2304 0p": { "VENC": 8 },
+    "4096*2304 15p": { "VENC": 8 }
+  }
+}
